graph/src/components/ethereum/types.rs

```diff
@@ -57,18 +57,44 @@ impl<'a, T> From<&'a Block<T>> for EthereumBlockData {
 #[derive(Clone, Debug)]
 pub struct EthereumTransactionData {
     pub hash: H256,
-    pub block_hash: H256,
-    pub block_number: U256,
+    pub index: U128,
+    pub from: H160,
+    pub to: H160,
+    pub value: U256,
     pub gas_used: U256,
+    pub gas_price: U256,
 }
 
 impl<'a> From<&'a Transaction> for EthereumTransactionData {
     fn from(tx: &'a Transaction) -> EthereumTransactionData {
         EthereumTransactionData {
             hash: tx.hash,
-            block_hash: tx.block_hash.unwrap(),
-            block_number: tx.block_number.unwrap(),
+            index: tx.transaction_index.unwrap(),
+            from: tx.from,
+            to: tx.to.unwrap_or(H160::from(0)),
+            value: tx.value,
             gas_used: tx.gas,
+            gas_price: tx.gas_price,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct EthereumLogData {
```

Comment:
Let's merge these fields into `EthereumEventData`. There's no need for duplicating the thing that represents the event.